sdk/core/core-httpclients/src/xhrHttpClient.ts

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortError } from "@azure/abort-controller";
+import {
+  type HttpClient,
+  type HttpHeaders,
+  type PipelineRequest,
+  type PipelineResponse,
+  type TransferProgressEvent,
+  createHttpHeaders,
+  RestError,
+} from "@azure/core-rest-pipeline";
+
+function isNodeReadableStream(x: unknown): x is NodeJS.ReadableStream {
```

Comment:
Q: why is this needed if this is browser only?

Reply:
TypeScript isn't smart enough to know?  These are needed to narrow the type of `body` to browser supported ones.